src/main/java/com/nutomic/syncthingandroid/fragments/DeviceSettingsFragment.java

```diff
@@ -142,6 +149,14 @@ else if (currentState != SyncthingService.State.ACTIVE) {
             }
             mDeviceId.setOnPreferenceClickListener(this);
         }
+
+        if (mDevice == null) {
+            // The device was removed from Syncthing and no reference was found
+            Log.d(TAG, "Empty mDevice");
+            getActivity().finish();
+            return;
+        }
```

Comment:
I guess that's right, we don't want a race condition. A lock might be more elegant though.